src/subscription_manager/cli_command/abstract_syspurpose.py

```diff
@@ -35,13 +35,19 @@
                         "by the entitlement server administrator.\n{advice}")
 SP_ADVICE = _("If you'd like to overwrite the server side change please run: {command}")
 
+# TRANSLATORS: this refers to a deprecated command
+DEPRECATED_COMMAND_MESSAGE = _("Deprecated, see 'syspurpose'")
+
 
 class AbstractSyspurposeCommand(CliCommand):
     """
     Abstract command for manipulating an attribute of system purpose.
     """
 
-    def __init__(self, name, shortdesc=None, primary=False, attr=None, commands=('set', 'unset', 'show', 'list')):
+    def __init__(self, name, subparser, shortdesc=None, primary=False, attr=None, commands=('set', 'unset', 'show', 'list')):
```

Comment:
I thought about it, and my idea was to force subclasses to provide a subparser. This way it is more explicit in case we add new syspurpose subcommands, which I guess we want only as subcommands of `syspurpose` and no more as top-level commands.